tests/console/zfs.pm

```diff
@@ -0,0 +1,190 @@
+# SUSE's openQA tests
+#
+# Copyright © 2020 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Tests the functionality of the zfs package
+# Maintainer: Felix Niederwanger <felix.niederwanger@suse.de>
+
+use base 'consoletest';
+use strict;
+use warnings;
+use testapi;
+use utils;
+use version_utils;
+use power_action_utils 'power_action';
+
+sub install_zfs {
+    # TODO: Replace -G (--no-gpgcheck) with something more sane
+    if (is_tumbleweed) {
+        zypper_call('addrepo -fG https://download.opensuse.org/repositories/filesystems/openSUSE_Tumbleweed/filesystems.repo');
+    } elsif (is_leap("=15.1")) {
+        zypper_call('addrepo -fG https://download.opensuse.org/repositories/filesystems/openSUSE_Leap_15.1/filesystems.repo');
+    } elsif (is_leap("=15.2")) {
+        zypper_call('addrepo -fG https://download.opensuse.org/repositories/filesystems/openSUSE_Leap_15.2/filesystems.repo');
+    } else {
+        die "Unsupported version";
+    }
+    zypper_call('refresh');
+    zypper_call('install zfs');
+}
+
+sub prepare_disks {
+    assert_script_run('fallocate -l 1GB /var/tmp/tank_a.img');
+    assert_script_run('fallocate -l 1GB /var/tmp/tank_b.img');
+    assert_script_run('fallocate -l 1GB /var/tmp/tank_c.img');
+    assert_script_run('fallocate -l 1GB /var/tmp/dozer.img');
+}
+
+sub corrupt_disk {
+    my $disk = shift;
+    assert_script_run("dd if=/dev/zero of=$disk bs=1G count=1");
+}
+
+sub cleanup {
+    script_run('cd');
+    script_run('if zpool list | grep tank ; then zpool destroy tank; fi');
+    script_run('if zpool list | grep dozer ; then zpool destroy dozer; fi');
+    script_run('rm -f /var/tmp/tank_{a,b,c}.img /var/tmp/dozer.img');
+}
+
+sub scrub {
+    my $pool = shift;
+    assert_script_run("zpool scrub $pool");
+    # Wait for scrub to finish
+    script_retry("zpool status $pool | grep scan | grep -v 'in progress'", delay => 1, retry => 30);
+}
+
+sub reboot {
+    my ($self) = @_;
+    power_action('reboot', textmode => 1);
+    $self->wait_boot(bootloader_time => 300);
+    select_console 'root-console';
+}
+
+sub run {
```

Comment:
I'd add some empty lines to separate each code block.

Reply:
Good idea.